Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -0,0 +1,168 @@
+import UIKit
+import PayPalNativeCheckout
+import PayPalCheckout
+import PaymentsCore
+import Card
+
+class PayPalViewModel: ObservableObject, PayPalDelegate {
+
+    enum State {
+        case initial
+        case loading(content: String)
+        case mainContent(title: String, content: String, flowComplete: Bool)
+    }
+
+    @Published private(set) var state = State.initial
+
+    private var accessToken = ""
+    private var payPalClient: PayPalClient?
+
+    func getAccessToken() {
+        Task {
+            state = .loading(content: "Getting access token")
+            guard let token = await getAccessToken() else {
+                state = .mainContent(title: "Error", content: "Failed to fetch access token", flowComplete: true)
+
+                return
+            }
+            accessToken = token
+            payPalClient = PayPalClient(config: CoreConfig(accessToken: token, environment: PaymentsCore.Environment.sandbox))
+            payPalClient?.delegate = self
+            state = .mainContent(title: "Access Token", content: accessToken, flowComplete: false)
+        }
+    }
+
+    func retry() {
+        payPalClient = nil
+        accessToken = ""
+        state = .initial
+    }
+
+    func checkoutWithOrder() {
```

Comment:
```suggestion
    func checkout(with order: Order) {
```
🧶: Same here for Swiftiness, though i'm not sure if `Order` is the correct type.

Reply:
same as above!